library/std/src/process/tests.rs

```diff
@@ -417,6 +417,111 @@ fn env_empty() {
     assert!(p.is_ok());
 }
 
+#[test]
+#[cfg(not(windows))]
+#[cfg_attr(any(target_os = "emscripten", target_env = "sgx"), ignore)]
+fn main() {
+    const PIDFD: &'static str =
+        if cfg!(target_os = "linux") { "    create_pidfd: false,\n" } else { "" };
+
+    let mut command = Command::new("some-boring-name");
+
+    assert_eq!(format!("{command:?}"), format!(r#""some-boring-name""#));
+
+    assert_eq!(
+        format!("{command:#?}"),
+        format!(
+            r#"Command {{
+    program: "some-boring-name",
+    args: [
+        "some-boring-name",
+    ],
+    env: CommandEnv {{
+        clear: false,
+        saw_path: false,
+        vars: {{}},
+    }},
+    cwd: None,
+    uid: None,
+    gid: None,
+    saw_nul: false,
+    groups: None,
+    pgroup: None,
+{PIDFD}}}"#
+        )
+    );
+
+    command.args(&["1", "2", "3"]);
+
+    assert_eq!(format!("{command:?}"), format!(r#""some-boring-name" "1" "2" "3""#));
+
+    assert_eq!(
+        format!("{command:#?}"),
+        format!(
+            r#"Command {{
+    program: "some-boring-name",
+    args: [
+        "some-boring-name",
+        "1",
+        "2",
+        "3",
+    ],
+    env: CommandEnv {{
+        clear: false,
+        saw_path: false,
+        vars: {{}},
+    }},
+    cwd: None,
+    uid: None,
+    gid: None,
+    saw_nul: false,
+    groups: None,
+    pgroup: None,
+{PIDFD}}}"#
+        )
+    );
+
+    crate::os::unix::process::CommandExt::arg0(&mut command, "exciting-name");
+
+    assert_eq!(
+        format!("{command:?}"),
+        format!(r#"["some-boring-name"] "exciting-name" "1" "2" "3""#)
+    );
+
+    assert_eq!(
+        format!("{command:#?}"),
+        format!(
+            r#"Command {{
+    program: "some-boring-name",
+    args: [
+        "exciting-name",
+        "1",
+        "2",
+        "3",
+    ],
+    env: CommandEnv {{
+        clear: false,
+        saw_path: false,
+        vars: {{}},
+    }},
+    cwd: None,
+    uid: None,
+    gid: None,
```

Comment:
Similar to the stdio stuff these can also be omitted when they have their default values, they're quite noisy.

Reply:
done